make peginconfirmationdepth a chain param

instagibbs opened the pull request.

jtimon commented:
Perhaps move the documentation for "-bitcoinconfirmationdepth" from init.cpp to https://github.com/ElementsProject/elements/blob/elements-0.14.1/src/chainparamsbase.cpp#L24 ? (oops, just noticed the -chainpetname argument isn't used anymore since its taken directly from -chain).

Bikeshed: perhaps pegin_min_depth or pegin_depth ?

Bikeshed: perhaps change -bitcoinconfirmationdepth to -con_min_depth (or -con_whateverthefinalname) for consistency?

Besides nits, utACK

instagibbs commented:
rebased, addressed the variable naming concern. It may be something we want to tweak in the future even on a production system, so not sure we should name it `-con_*`. It's almost not consensus, in the sense we can set it depending on how we feel about the parent chain's stability.
